File: python/tests/test_config.py
"""Test configuration utility for loading test settings from config file.
Supports environment variable overrides.
"""

import os
import configparser
from pathlib import Path


class TestConfig:
    """Test configuration loader with environment variable support."""
    
    _config = None
    _config_file = Path(__file__).parent / "test-config.ini"
    
    @classmethod
    def _load_config(cls):
        """Load configuration from file."""
        if cls._config is not None:
            return cls._config
        
        cls._config = configparser.ConfigParser()
        if cls._config_file.exists():
            cls._config.read(cls._config_file)
        return cls._config
    
    @classmethod
    def get_registry_name(cls):
        """Get registry name with environment variable override."""
        env_value = os.getenv("GLUE_REGISTRY_NAME")
        if env_value:
            return env_value
        
        config = cls._load_config()
        return config.get("test", "glue.registry.name", fallback="glue-schema-registry-ansumanroy-6219")
    
    @classmethod
    def get_aws_region(cls):
        """Get AWS region with environment variable override."""
        env_value = os.getenv("AWS_REGION")
        if env_value:
            return env_value
        
        config = cls._load_config()
        return config.get("test", "aws.region", fallback="us-east-1")
    
    @classmethod
    def get_avro_schema_name(cls):
        """Get Avro schema name with environment variable override."""
        env_value = os.getenv("SCHEMA_NAME_AVRO")
        if env_value:
            return env_value
        
        config = cls._load_config()
        return config.get("test", "schema.name.avro", fallback="SalesforceAudit")
    
    @classmethod
    def get_json_schema_name(cls):
        """Get JSON schema name with environment variable override."""
        env_value = os.getenv("SCHEMA_NAME_JSON")
        if env_value:
            return env_value
        
        config = cls._load_config()
        return config.get("test", "schema.name.json", fallback="SalesAuditJSON")

File: java/src/test/java/com/aws/glue/schema/registry/config/TestConfig.java
package com.aws.glue.schema.registry.config;

import software.amazon.awssdk.regions.Region;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

/**
 * Utility class for loading test configuration from resource file.
 * Supports environment variable overrides.
 */
public class TestConfig {
    
    private static final String CONFIG_FILE = "/test-config.properties";
    private static Properties properties;
    
    static {
        loadProperties();
    }
    
    private static void loadProperties() {
        properties = new Properties();
        try (InputStream inputStream = TestConfig.class.getResourceAsStream(CONFIG_FILE)) {
            if (inputStream != null) {
                properties.load(inputStream);
            }
        } catch (IOException e) {
            // If config file is not found, use defaults
            System.err.println("Warning: Could not load test-config.properties, using defaults");
        }
    }
    
    /**
     * Gets the registry name, with environment variable override support.
     * 
     * @return Registry name
     */
    public static String getRegistryName() {
        String envValue = System.getenv("GLUE_REGISTRY_NAME");
        if (envValue != null && !envValue.isEmpty()) {
            return envValue;
        }
        String propValue = System.getProperty("glue.registry.name");
        if (propValue != null && !propValue.isEmpty()) {
            return propValue;
        }
        return properties.getProperty("glue.registry.name", "glue-schema-registry-ansumanroy-6219");
    }
    
    /**
     * Gets the AWS region, with environment variable override support.
     * 
     * @return AWS region
     */
    public static Region getAWSRegion() {
        String envValue = System.getenv("AWS_REGION");
        if (envValue != null && !envValue.isEmpty()) {
            return Region.of(envValue);
        }
        String propValue = System.getProperty("aws.region");
        if (propValue != null && !propValue.isEmpty()) {
            return Region.of(propValue);
        }
        String regionStr = properties.getProperty("aws.region", "us-east-1");
        return Region.of(regionStr);
    }
    
    /**
     * Gets the Avro schema name.
     * 
     * @return Avro schema name
     */
    public static String getAvroSchemaName() {
        String envValue = System.getenv("SCHEMA_NAME_AVRO");
        if (envValue != null && !envValue.isEmpty()) {
            return envValue;
        }
        return properties.getProperty("schema.name.avro", "SalesforceAudit");
    }
    
    /**
     * Gets the JSON schema name.
     * 
     * @return JSON schema name
     */
    public static String getJsonSchemaName() {
        String envValue = System.getenv("SCHEMA_NAME_JSON");
        if (envValue != null && !envValue.isEmpty()) {
            return envValue;
        }
        return properties.getProperty("schema.name.json", "SalesAuditJSON");
    }
}

